pkg/cmd/server/admin/decrypt.go

```diff
@@ -0,0 +1,158 @@
+package admin
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/openshift/origin/pkg/cmd/util"
+	"github.com/spf13/cobra"
+
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	pemutil "github.com/openshift/origin/pkg/cmd/util/pem"
+)
+
+const DecryptCommandName = "decrypt"
+
+type DecryptOptions struct {
+	Data    *pem.Block
+	InFile  string
+	OutFile string
+
+	Key       []byte
+	KeyArg    string
```

Comment:
Looks like `Key` is the value you want in your options struct (needed for the operation) and `KeyArg` doesn't belong here since its only used by this command to get a key: (not likely to be used in a delegation usage).